src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -1424,17 +1450,17 @@ public static void shutdown() {
         service.preShutdown();
       }
 
-      /*
-       * - huge amount of json that is not used for (ServiceInterface service :
-       * getServices()) { service.save(); }
-       */
-
       log.info("releasing all");
+
+      // release
       releaseAll();
     } catch (Exception e) {
       log.error("something threw - continuing to shutdown", e);
     }
 
+    // calling System.exit(0) before some specialized threads
+    // are completed will actually end up in a deadlock
+    Service.sleep(1000);

```

Comment:
This seems bad.. why would system.exit(0) cause a deadlock?

Reply:
not sure .. but there's a lot of deps & code involved ... mebbe its only eclipse related and their remote tooling ... dunno